### PR TITLE
Add mqttv5 subscription options supported.

### PR DESCRIPTION
**Pull request by wanghaEMQ**

### CODE REVIEW

File: src/supplemental/mqtt/mqtt_codec.c [hunk 1]
```diff
@@ -1987,6 +1993,8 @@ nni_mqttv5_msg_decode_subscribe(nni_msg *msg)
 	uint16_t temp_length       = 0;
 	uint32_t topic_count       = 0;
 
+	uint8_t to; // topic option
```
Comment (alvin1221) [inline src/supplemental/mqtt/mqtt_codec.c:1996]: You'd better to initailize this `to`

**Review by alvin1221 — APPROVED** [review]

**Review by lee-emqx — APPROVED** [review]

File: include/nng/mqtt/mqtt_client.h [hunk 2]
```diff
@@ -476,8 +479,8 @@ NNG_DECL nng_mqtt_topic *nng_mqtt_topic_array_create(size_t);
 NNG_DECL void nng_mqtt_topic_array_set(nng_mqtt_topic *, size_t, const char *);
 NNG_DECL void nng_mqtt_topic_array_free(nng_mqtt_topic *, size_t);
 NNG_DECL nng_mqtt_topic_qos *nng_mqtt_topic_qos_array_create(size_t);
-NNG_DECL void                nng_mqtt_topic_qos_array_set(
-                   nng_mqtt_topic_qos *, size_t, const char *, uint8_t);
+NNG_DECL void nng_mqtt_topic_qos_array_set(nng_mqtt_topic_qos *, size_t,
+           const char *, uint8_t, uint8_t, uint8_t, uint8_t);
```
Comment (JaylinYu) [inline include/nng/mqtt/mqtt_client.h:483]: we should do the same for mqtt v5 bridigng

**Review by JaylinYu — APPROVED** [review]

File: src/supplemental/mqtt/mqtt_codec.c (hunk 1 above)
Comment (JaylinYu) [reply src/supplemental/mqtt/mqtt_codec.c:1996]: I will fix this later

File: include/nng/mqtt/mqtt_client.h (hunk 2 above)
Comment (wanghaEMQ) [reply include/nng/mqtt/mqtt_client.h:483]: I agree

**Pull request merged**